src/cobra/sampling/hr_sampler.py

```diff
@@ -201,9 +189,48 @@ def __init__(self, model, thinning, nproj=None, seed=None):
         # Avoid overflow
         self._seed = self._seed % np.iinfo(np.int32).max
 
-    def __build_problem(self):
-        """Build the matrix representation of the sampling problem."""
+    def __build_problem(self) -> NamedTuple:
+        """Build the matrix representation of the sampling problem.
 
+        Returns
+        -------
+        NamedTuple
+            A named tuple consisting of 6 arrays, 1 matrix and 1 boolean:
+            - "equalities" is a matrix `S` such that `S * vars = b`. It
+              includes a row for each constraint and one column for each
+              variable.
+            - "b" is the right side of the equality equation such that
+              `S * vars = b`.
+            - "inequalities" is a matrix M such that `lb <= M * vars <= ub`.
+              It contains a row for each inequality and as many columns as
+              variables.
+            - "bounds" is a compound matrix [lb ub] containing the lower and
+              upper bounds for the inequality constraints in M.
+            - "variable_fixed" is a boolean vector indicating whether the
+              variable at that index is fixed (`lower bound == upper_bound`)
+              and is thus bounded by an equality constraint.
+            - "variable_bounds" is a compound matrix `[lb ub]` containing the
+              lower and upper bounds for all variables.
+            - "nullspace" is a matrix containing the nullspace of the equality
+              constraints. Each column is one basis vector.
+            - "homogeneous" is a boolean which indicates whether the sampling
+              problem is homogeneous, e.g., whether there exists no non-zero
+              fixed variables or constraint.
+
+        """
+        Problem = NamedTuple(
```

Comment:
This won't work. This hast to be defined outside of a class so you can pickle it. See comment on step() below.

Reply:
Okay I see. A strange is that the tests are passing for Python 3.6 and 3.7 and is similar on my local system too. Any idea why it's not failing?

Reply:
On Linux, they should pass because of fork vs. spawn when running multiprocessing. The multiprocess backend changed quite a bit between minor versions on Mac and Windows so I would not be too surprised about it working on one version and not the other.

Reply:
That's a fair point. Okay I'll revert it back.

Reply:
Done.